src/framework/tours/view/toursprovider.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2025 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "toursprovider.h"
+
+#include "log.h"
+
+using namespace muse;
+using namespace muse::tours;
+
+ToursProvider::ToursProvider(const modularity::ContextPtr& iocCtx)
+    : QObject(), Injectable(iocCtx)
+{
+    QObject::connect(&m_openTimer, &QTimer::timeout, this, &ToursProvider::doShow);
+}
+
+void ToursProvider::showTour(const TourStepMetaList& steps)
+{
+    m_steps = steps;
+    m_totalSteps = m_steps.size();
+
+    showNext();
+}
+
+void ToursProvider::showNext()
+{
+    m_openTimer.start();
+}
+
+void ToursProvider::doShow()
+{
+    m_openTimer.stop();
+
+    const TourStepMeta& meta = muse::takeFirst(m_steps);
+    int index = m_totalSteps - m_steps.size();
+
+    QQuickItem* parentItem = findControl(meta.controlName);
+    IF_ASSERT_FAILED(parentItem) {
+        return;
+    }
+
+    emit openTourStep(parentItem, meta.title, meta.description, meta.videoExplanationUrl, index, m_totalSteps);
+}
+
+QQuickItem* ToursProvider::findControl(const String& controlName)
```

Comment:
I liked the idea of ​​searching for controls by key: section, panel, control...
This will allow us not to set names for controls as uris, which will essentially duplicate the information already available